### PR TITLE
Updated dependency versions aws-kms-pkcs11 & aws-sdk-cpp

### PR DESCRIPTION
**Pull request by sumukhballal**
**Description of changes:**

- Updated aws-kms-pkcs11 version to 0.0.11 
- Update aws-sdk-cpp version to 1.11.398
- Updates hashes for aws-sdk-cpp to the latest version.
- Updates Go1.22 to 1.22.7.
- Updates Go1.23 to 1.23.1.

**Testing done:**

 - [x] make (on both x86 and aarch64 hosts)
 - [x] docker run
 - [x] Built bottlerocket core kit on aarch64 host
 - [x] Build bottlerocket core kit on x86 host

Note: We now also build the SDK for ACM-PCA because the new version of aws-kms-pkcs11 v0.0.11 has a ACM-PCA dependency.

**Terms of contribution:**

By submitting this pull request, I agree that this contribution is dual-licensed under the terms of both the Apache License, version 2.0, and the MIT license.

### CODE REVIEW

File: hashes/aws-sdk-cpp [hunk 1]
```diff
@@ -1,31 +1,31 @@
 ### See https://github.com/aws/aws-sdk-cpp/blob/1.11.207/prefetch_crt_dependency.sh for what to use below. ###
```
Comment (bcressey) [inline hashes/aws-sdk-cpp]:
```suggestion
### See https://github.com/aws/aws-sdk-cpp/blob/1.11.398/prefetch_crt_dependency.sh for what to use below. ###
```

File: hashes/aws-sdk-cpp [hunk 2]
```diff
@@ -1,31 +1,31 @@
...
-SHA512 (aws-checksums.tar.gz) = 297f0c75f01a7560a55fe30a9a56fb885beef90dc44b952d4e66ca70130ae6ef0da62d2fec3fc9aaedf6c26bc39a072cd36e8668321045bc7695e8f6ddec2e37
-# https://codeload.github.com/awslabs/aws-lc/tar.gz/a8d06de79e405692ac06fe17163626eaab515e4e
-SHA512 (aws-lc.tar.gz) = 005ab1d1c80885e10c8dd381318dea0cfb2e017afa906ca84900d4a96661adf782b2d06c73ccfbacce7100b3888ace4f6344e684bc3cde91f66b295e327d4168
-# https://codeload.github.com/awslabs/s2n/tar.gz/95753f0c528b59025343e8799cb25d3e9df89e21
-SHA512 (s2n.tar.gz) = f9d5fb7db71f09f61ae791457a3ba6392f014e820e2e83875f88105e921ff08daef6ba82ee2e767de652a3a7297016d1ab9fc54e5369e8e747b43af7d53174a9
-# https://github.com/awslabs/aws-sdk-cpp/archive/1.11.207.tar.gz#/aws-sdk-cpp-1.11.207.tar.gz
-SHA512 (aws-sdk-cpp-1.11.207.tar.gz) = 8b6238f7eefb1be8c047489113c4bf285ef99fae7270bd38581a5488c0b1e0b7d69b45106daf1db45aabbc93b636113cab673340c82f1c5e3191f25d6ad3d3a5
+# https://codeload.github.com/awslabs/aws-crt-cpp/tar.gz/e14c814728dc438a0b7dc694bd83418addc5a6a1
+SHA512 (aws-crt-cpp.tar.gz) = 863d0a76593a266dbfb53d2f23fbfefc574bd502deddeaec622716db27cf0d519fb2140bfde0de3a391d657f205b0c6fa1a98499382d2e4f531424dccc408696
```
Comment (bcressey) [inline hashes/aws-sdk-cpp]: All of the other hashes match, but I get this for `aws-crt-cpp.tar.gz`:
```
1b06204d42a7ecd2611329be54769bc470444311da596b93fc48cf710d3670d292bcd77a0315392ff45617994f0d713dcd028ca6781841e741f3c40e48be2ce7
```

**Comment by sumukhballal** [comment]
Addressed Ben's comment and added Go updates.

**Comment by sumukhballal** [comment]
Updated Go 1.23 to v1.23.1!

**Review by ginglis13 — APPROVED** [review]
The Go updates LGTM

**Review by bcressey — APPROVED** [review]

**Pull request merged**